WowPacketParserModule.V9_0_1_36216/Parsers/QuestHandler.cs

```diff
@@ -441,9 +442,10 @@ public static void QuestGiverOfferReward(Packet packet)
         [Parser(Opcode.CMSG_QUEST_GIVER_CHOOSE_REWARD)]
         public static void HandleQuestChooseReward(Packet packet)
         {
-            packet.ReadPackedGuid128("QuestGiverGUID");
-            packet.ReadInt32("QuestID");
-            ReadRewardItem(packet, "ItemChoice");
+            var chooseReward = packet.Holder.ClientQuestGiverChooseReward = new();
+            chooseReward.QuestGiver = packet.ReadPackedGuid128("QuestGiverGUID");
+            chooseReward.QuestId = packet.ReadUInt32("QuestID");
```

Comment:
not sure if we should do it like this; you changed the sign here

Reply:
Indeed, in most handlers quest id was read as uint, hence the common structure has uint, and I've thought it probably should be uint here. I guess it doesn't matter in case of quest id, because no quest id is (and never will be) higher than max int32. But I can revert this change and just cast the value from int to uint.

Reply:
i'd stick to client sign and cast for such purposes, would suggest to wait on another opinion though

Reply:
Fixed the sign